### ISSUE TITLE
[BUG] Fatal Error in Java Runtime Environment when Initiating WebRTC Remote Connection to Controller

### ISSUE THREAD

**Issue by Junto026** [linuxserver/docker-unifi-controller#207]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The fatal error occurs when initiating a remote access connection (either from network.unifi.ui.com using WebRTC or from a mobile app which uses WebRTC).  When the connection is attempted, the below output is displayed on the container logs.  The remote connection fails, and re-attempting results in the error and output being repeated.

> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:11 :118 TCP candidates not supported yet
> 2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
> 2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
> 2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
> 2023-04-22 10:25:12 :484 All TURN channel bindings failed
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 # A fatal error has been detected by the Java Runtime Environment:
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 #  SIGILL (0x4) at pc=0x0000fffea9c00760, pid=153, tid=328
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 # JRE version: OpenJDK Runtime Environment (11.0.18+10) (build 11.0.18+10-post-Ubuntu-0ubuntu120.04.1)
> 2023-04-22 10:25:12 # Java VM: OpenJDK 64-Bit Server VM (11.0.18+10-post-Ubuntu-0ubuntu120.04.1, mixed mode, tiered, compressed oops, g1 gc, linux-aarch64)
> 2023-04-22 10:25:12 # Problematic frame:
> 2023-04-22 10:25:12 # C  [libubnt_webrtc_jni.so+0x2f5760]
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 # An error report file with more information is saved as:
> 2023-04-22 10:25:12 # /config/hs_err_pid153.log
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:12 # If you would like to submit a bug report, please visit:
> 2023-04-22 10:25:12 #   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
> 2023-04-22 10:25:12 # The crash happened outside the Java Virtual Machine in native code.
> 2023-04-22 10:25:12 # See problematic frame for where to report the bug.
> 2023-04-22 10:25:12 #
> 2023-04-22 10:25:13 WARNING: An illegal reflective access operation has occurred
> 2023-04-22 10:25:13 WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
> 2023-04-22 10:25:13 WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
> 2023-04-22 10:25:13 WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
> 2023-04-22 10:25:13 WARNING: All illegal access operations will be denied in a future release

### Expected Behavior

The remote connection should establish and the controller should be accessible through the mobile app or network.unifi.ui.com.

### Steps To Reproduce

1. Build a Unifi Controller in Docker on Mac (ARM Architecture) with version 7.3.83-ls189 Build-date:- 2023-04-21T02:39:54+00:00

2. Configure Remote Access.  I did this by reverting to the legacy UI and then enabling at Settings --> Remote Access --> Enable Remote Access.

3. Confirm Remote Access Status shows "Connected"

4. Log into network.unifi.ui.com, select the controller, and launch remote access to it (ensure you're using a WebRTC compatible browser, such as Chrome, and you select the WebRTC connection method option.

### Environment

```markdown
- OS:Docker on MacOS (M2 Chip / ARM Architecture)
- How docker service was installed:
docker-compose -f './unifi.yml' pull && docker-compose -f './unifi.yml' up --force-recreate --build -d;


docker inspect -f '{{ index .Config.Labels "build_version" }}' unifi
Linuxserver.io version:- 7.3.83-ls189 Build-date:- 2023-04-21T02:39:54+00:00

docker inspect -f '{{ index .Config.Labels "build_version" }}' lscr.io/linuxserver/unifi-controller:latest
Linuxserver.io version:- 7.3.83-ls189 Build-date:- 2023-04-21T02:39:54+00:00
```

### CPU architecture

arm64

### Docker creation

```bash
unifi.yml file:


volumes: 
 unifi-data: 
  external: true
services:
 unifi:
  image: lscr.io/linuxserver/unifi-controller:latest
  container_name: unifi
  volumes:
   - unifi-data:/config
  environment:
   - PUID=501
   - PGID=20
   - TZ=America/New_York
  ports:
   - 8080:8080
   - 26000:8443
   - 3478:3478/udp
   - 10001:10001/udp
  restart: always
```

### Container logs

```bash
2023-04-22 10:19:39 [migrations] started
2023-04-22 10:19:39 [migrations] no migrations found
2023-04-22 10:19:39 ───────────────────────────────────────
2023-04-22 10:19:39 
2023-04-22 10:19:39       ██╗     ███████╗██╗ ██████╗ 
2023-04-22 10:19:39       ██║     ██╔════╝██║██╔═══██╗
2023-04-22 10:19:39       ██║     ███████╗██║██║   ██║
2023-04-22 10:19:39       ██║     ╚════██║██║██║   ██║
2023-04-22 10:19:39       ███████╗███████║██║╚██████╔╝
2023-04-22 10:19:39       ╚══════╝╚══════╝╚═╝ ╚═════╝ 
2023-04-22 10:19:39 
2023-04-22 10:19:39    Brought to you by linuxserver.io
2023-04-22 10:19:39 ───────────────────────────────────────
2023-04-22 10:19:39 
2023-04-22 10:19:39 To support LSIO projects visit:
2023-04-22 10:19:39 https://www.linuxserver.io/donate/
2023-04-22 10:19:39 
2023-04-22 10:19:39 ───────────────────────────────────────
2023-04-22 10:19:39 GID/UID
2023-04-22 10:19:39 ───────────────────────────────────────
2023-04-22 10:19:39 
2023-04-22 10:19:39 User UID:    501
2023-04-22 10:19:39 User GID:    20
2023-04-22 10:19:39 ───────────────────────────────────────
2023-04-22 10:19:39 
2023-04-22 10:19:39 [custom-init] No custom files found, skipping...
2023-04-22 10:19:51 [ls.io-init] done.
2023-04-22 10:19:40 WARNING: An illegal reflective access operation has occurred
2023-04-22 10:19:40 WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
2023-04-22 10:19:40 WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
2023-04-22 10:19:40 WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
2023-04-22 10:19:40 WARNING: All illegal access operations will be denied in a future release
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:11 :118 TCP candidates not supported yet
2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:25:12 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:25:12 :484 All TURN channel bindings failed
2023-04-22 10:25:12 #
2023-04-22 10:25:12 # A fatal error has been detected by the Java Runtime Environment:
2023-04-22 10:25:12 #
2023-04-22 10:25:12 #  SIGILL (0x4) at pc=0x0000fffea9c00760, pid=153, tid=328
2023-04-22 10:25:12 #
2023-04-22 10:25:12 # JRE version: OpenJDK Runtime Environment (11.0.18+10) (build 11.0.18+10-post-Ubuntu-0ubuntu120.04.1)
2023-04-22 10:25:12 # Java VM: OpenJDK 64-Bit Server VM (11.0.18+10-post-Ubuntu-0ubuntu120.04.1, mixed mode, tiered, compressed oops, g1 gc, linux-aarch64)
2023-04-22 10:25:12 # Problematic frame:
2023-04-22 10:25:12 # C  [libubnt_webrtc_jni.so+0x2f5760]
2023-04-22 10:25:12 #
2023-04-22 10:25:12 # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
2023-04-22 10:25:12 #
2023-04-22 10:25:12 # An error report file with more information is saved as:
2023-04-22 10:25:12 # /config/hs_err_pid153.log
2023-04-22 10:25:12 #
2023-04-22 10:25:12 # If you would like to submit a bug report, please visit:
2023-04-22 10:25:12 #   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
2023-04-22 10:25:12 # The crash happened outside the Java Virtual Machine in native code.
2023-04-22 10:25:12 # See problematic frame for where to report the bug.
2023-04-22 10:25:12 #
2023-04-22 10:25:13 WARNING: An illegal reflective access operation has occurred
2023-04-22 10:25:13 WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
2023-04-22 10:25:13 WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
2023-04-22 10:25:13 WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
2023-04-22 10:25:13 WARNING: All illegal access operations will be denied in a future release
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :118 TCP candidates not supported yet
2023-04-22 10:26:57 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:26:57 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:26:57 :1023 Permanent error code on channel bind request: 403 - Forbidden IP
2023-04-22 10:26:57 :484 All TURN channel bindings failed
2023-04-22 10:26:57 #
2023-04-22 10:26:57 # A fatal error has been detected by the Java Runtime Environment:
2023-04-22 10:26:57 #
2023-04-22 10:26:57 #  SIGILL (0x4) at pc=0x0000fffeac802760, pid=455, tid=631
2023-04-22 10:26:57 #
2023-04-22 10:26:57 # JRE version: OpenJDK Runtime Environment (11.0.18+10) (build 11.0.18+10-post-Ubuntu-0ubuntu120.04.1)
2023-04-22 10:26:57 # Java VM: OpenJDK 64-Bit Server VM (11.0.18+10-post-Ubuntu-0ubuntu120.04.1, mixed mode, tiered, compressed oops, g1 gc, linux-aarch64)
2023-04-22 10:26:57 # Problematic frame:
2023-04-22 10:26:57 # C  [libubnt_webrtc_jni.so+0x2f5760]
2023-04-22 10:26:57 #
2023-04-22 10:26:57 # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
2023-04-22 10:26:57 #
2023-04-22 10:26:57 # An error report file with more information is saved as:
2023-04-22 10:26:57 # /config/hs_err_pid455.log
2023-04-22 10:26:57 #
2023-04-22 10:26:57 # If you would like to submit a bug report, please visit:
2023-04-22 10:26:57 #   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
2023-04-22 10:26:57 # The crash happened outside the Java Virtual Machine in native code.
2023-04-22 10:26:57 # See problematic frame for where to report the bug.
2023-04-22 10:26:57 #
2023-04-22 10:26:58 WARNING: An illegal reflective access operation has occurred
2023-04-22 10:26:58 WARNING: Illegal reflective access by org.apache.catalina.loader.WebappClassLoaderBase (file:/usr/lib/unifi/lib/tomcat-embed-core-9.0.59.jar) to field java.io.ObjectStreamClass$Caches.localDescs
2023-04-22 10:26:58 WARNING: Please consider reporting this to the maintainers of org.apache.catalina.loader.WebappClassLoaderBase
2023-04-22 10:26:58 WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
2023-04-22 10:26:58 WARNING: All illegal access operations will be denied in a future release
```

**Comment by j0nnymoe** [comment]
I suspect this issue is being caused in the differences between docker engine on Linux and having to use docker desktop on a Mac.

Could you test on a Linux host to see if you have the same outcome?
 
MacOS support (along with windows) is best effort for us as no one within the team uses these platforms to deploy containers (and they're some what limited anyways)

**Comment by Junto026** [comment]
I will try and let you know. FWIW, this was working up until recently.  I’m not sure exactly which update version the issue started in, but my container updates weekly so it must have been within the last few weeks. JacobOn Apr 22, 2023, at 1:37 PM, j0nnymoe ***@***.***> wrote:﻿
I suspect this issue is being caused in the differences between docker engine on Linux and having to use docker desktop on a Mac.
Could you test on a Linux host to see if you have the same outcome?
MacOS support (along with windows) is best effort for us as no one within the team uses these platforms to deploy containers (and they're some what limited anyways)

**Comment by Junto026** [comment]
Interestingly, I tested previous images (7.3.76, 7.2.95, and 7.2.94) and the fatal error occurred with all of those versions.  So I am not sure how I was using the remote access feature up until recently.  I also tried switching between gRPC FUSE and VirtioFS for the file sharing implementation, which had no impact.

Perhaps an update to the docker engine / docker desktop for Mac triggered the issue?  I am running Docker Desktop 4.18.0 and Docker Engine v20.1
0.24.

Finally, attached is a much more detailed crash report file.

[hs_err_pid141.log](https://github.com/linuxserver/docker-unifi-controller/files/11356429/hs_err_pid141.log)

**Comment by Junto026** [comment]
Commenting to keep the issue open, understanding Docker for MacOS is a low priority / best effort platform for support.

**Comment by mrlexley** [comment]
I face the same issue for multiple versions of the image.

**Comment by Junto026** [comment]
Commenting to keep the issue open, understanding Docker for MacOS is a low priority / best effort platform for support.

**Comment by drizuid** [comment]
This isn't a valid issue for us, we do not support or test docker desktop on any operating system. We do not support or test on MacOS. When we make changes, we make them assuming the users are using a debian derived linux system which CAN (not intentionally) break unsupported systems at any time. If you submit a PR that does not cause an issue for supported systems but fixes the issue for your unsupported system, we are willing to consider merging, but otherwise, we will not work on this issue.
